scripts/nbconverted/latent_space_comparison.py

```diff
@@ -0,0 +1,388 @@
+                             vae_tl300_ov_vector], axis=1)
+latent_space_df.head(2)
+
+
+# In[13]:
+
+# Process latent space dataframe to long format
+long_latent_df = latent_space_df.stack().reset_index()
+long_latent_df.columns = ['rank', 'algorithm', 'feature_activity']
+
+# Distinguish node activation by feature
+long_algorithms_df = long_latent_df[long_latent_df['algorithm'].isin(algorithms)]
+long_algorithms_df.reset_index(drop=True, inplace=True)
+
+long_features_df = long_latent_df[~long_latent_df['algorithm'].isin(algorithms)]
+long_features_df.reset_index(drop=True, inplace=True)
+
+# Concatenate node assignments to the dataframe
+long_latent_space_df = pd.concat([long_algorithms_df, long_features_df],
+                                 ignore_index=True, axis=1)
+long_latent_space_df.columns = ['rank', 'algorithm', 'activation', 'feature_rank',
+                                'feature_name', 'feature']
+long_latent_space_df.head(2)
+
+
+# In[14]:
+
+# Assign color to each algorithm
+long_latent_space_df = long_latent_space_df.assign(algorithm_color =
+                                                   long_latent_space_df['algorithm'])
+
+algorithm_color_dict = {'pca': '#a6cee3',
+                        'ica': '#1f78b4',
+                        'nmf': '#b2df8a',
+                        'adage': '#33a02c',
+                        'tybalt': '#fb9a99',
+                        'vae_100': '#e31a1c',
+                        'vae_300': '#fdbf6f'}
+
+long_latent_space_df = long_latent_space_df.replace({'algorithm_color': algorithm_color_dict})
+
+# Drop redundant columns
+long_latent_space_df = long_latent_space_df.drop(['feature_rank', 'feature_name'], axis=1)
+long_latent_space_df.head(2)
+
+
+# In[15]:
+
+# Output ranking and activation scores per feature per algorithm
+latent_output_file = os.path.join('results',
+                                  'hgsc_mesenchymal_immunoreactive_algorithm_subtract.tsv')
+long_latent_space_df.to_csv(latent_output_file, index=False, sep='\t')
+print(long_latent_space_df.shape)
+long_latent_space_df.head()
+
+
+# In[16]:
+
+latent_space_figure = os.path.join('figures', 'algorithm_comparison_latent_space.png')
+ax = sns.pointplot(x='rank', y='activation', hue='algorithm', data=long_latent_space_df,
+                   palette=algorithm_color_dict, markers=['x', '3', '4', 'd', '*', 'd','o'],
+                   orient='v', scale=0.6)
+ax.set_xlabel('Feature Rank')
+ax.set_ylabel('Mesenchymal - Immunoreactive\nFeature Activation')
+ax.set(xticklabels=[]);
+plt.tight_layout()
+plt.setp(ax.get_legend().get_texts(), fontsize='12')
+plt.setp(ax.get_legend().get_title(), fontsize='16')
+plt.savefig(latent_space_figure, dpi=600, height=6, width=5)
+
+
+# ### Part II. Extract high weight genes from each most explanatory feature
+
+# In[17]:
+
+def get_high_weight_genes(weight_matrix, node, algorithm, high_std=2.5, direction='positive',
+                          output_file=False):
```

Comment:
Would change output_file to empty quotes i.e. `''`. Otherwise someone would thing that this is a boolean variable and not a string.

Reply:
good point - I will update in next commit

Reply:
updated in 8de18c7d69ecf1f7da92e1e0d6a2597bde1bdf14 thanks!